src/collection.js

```diff
@@ -610,6 +610,21 @@ export default class Collection {
     );
   }
 
+  /**
+   * Same as {@link Collection#deleteAll}, but wrapped in its own transaction, execulding the parameter.
+   *
+   * @return {Promise}
+   */
+  async deleteAll() {
+    const { data } = await this.list({}, { includeDeleted: false });
+    return this.execute(
+      transaction => {
+        return transaction.deleteAll(data.map(record => record.id));
+      },
+      { preloadIds: data.map(record => record.id) }
```

Comment:
nitpick: avoid iterating twice (store in variable like `recordIds`)

Reply:
Good spot, map is too useful and easy to use I didn't even notice it 😄